### ISSUE TITLE
Add ignore: ["duplicates-with-prefixed-values"] option to declaration-block-no-duplicate-properties

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#882]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

The example CSS below detects `width` as duplicated properties.

> Which rule, if any, is this issue related to? 

`declaration-block-no-duplicate-properties`

> What CSS is needed to reproduce this issue?

```css
a { 
	width: -webkit-calc(47.6% - 4px);
	width: calc(47.6% - 4px);
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "declaration-block-no-duplicate-properties": true,
  }
}
```

Note: No stylelint vendor prefixes are enabled, hence, at this stage I have lots of vendor prefixes throughout the CSS used in this testing.

> Which version of stylelint are you using? 

`4.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

grunt-postcss 

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)? No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
/css/test.css
151:2	⚠  Unexpected duplicate property "width" (declaration-block-no-duplicate-properties) [stylelint]
```

**Comment by jeddy3** [comment]
I think the rule is behaving as expected.

One of the original reasons we added the rule was to encourage tooling e.g. using [`autoprefixer`](https://github.com/postcss/autoprefixer) for this instance or things like [`pixrem`](https://github.com/robwierzbowski/node-pixrem) if you need to add a fallback:

```css
a { font-size: 2rem; }
```

Instead of:

```css
a { font-size: 32px; font-size: 2rem; }
```

There is maybe scope for adding an option to inspect the value of the properties for things like vendor prefixes, but this will add complexity (to a simple rule) and I think would be a slippery slope. 

If you're unable to use the likes of autoprefixer on your source then I guess the recommendation would be to turn off the rule for now.

We can update the rule's documentation to explain this a bit better (similar to how we do in [`no-unsupported-browser-features`](https://github.com/stylelint/stylelint/tree/master/src/rules/no-unsupported-browser-features#no-unsupported-browser-features).

**Comment by ntwb** [comment]
WordPress does actually use autoprefixer, the difference is in that the vendor prefixes added or removed by autoprefixer are committed to the code base. 

Typically projects use a `/dist` folder to where vendor prefixes would be added by a task runner along with concatenation and minification etc, WordPress only performs this for CSS minifcation.

WordPress does this primarily so people can hack on it out of the box in any browser of choice and not having to have people know how to install build tools (NodeJS, NPM etc) to make it work.

I've kind of been think about a `no-unsupported-vendor-prefixes` rule that would whitelist the vendor prefixes based on the browser list array passed to `browserlist`, in fact very similar to the `no-unsupported-browser-features` rule.
 
Let me think about all of this some more...

**Comment by davidtheclark** [comment]
One abstract idea I have that might address your problem @ntwb is to see if we could add to `declaration-block-no-duplicate-properties` a new option -- `allow-duplicates-with-prefixed-values` or something (better name?) -- that would do something like this:

- If duplicate properties are found, compare the values of each after stripping vendor prefixes with `require('postcss/lib/vendor').unprefixed` (or some other Autoprefixer internal --- or maybe we have a util that does something appropriate ...).
- If the values are identical after stripping vendor prefixes, don't complain.
- If they differ after stripping vendor prefixes, DO complain, because it's probably a mistake.

**Comment by jeddy3** [comment]
Sgtm.

It'd be a boolean option right? So, `ignoreDuplicatesWithPrefixedValues: true`.

**Comment by jeddy3** [comment]
In case there's still an desire to implement this... I just realised that my suggested option name goes against our existing conventions and I pretty sure this should just a normal `ignore` keyword option e.g.

`ignore: ["duplicates-with-prefixed-values"]`

As we might want to add ignore more things in the future.

Anyway, marking this an "help wanted" and "option" as that's how this issue can be resolved.

**Comment by jeddy3** [comment]
Closing as the `ignore: ["consecutive-duplicates"]` option makes this kind of redundant.
